ablySpec/Auth.swift

```diff
@@ -599,5 +599,164 @@ class Auth : QuickSpec {
+
+                    expect(serverTime).toNot(beNil(), description: "Server time is nil")
+
+                    let authOptions = ARTAuthOptions()
+                    authOptions.queryTime = true
+
+                    rest.auth.createTokenRequest(nil, options: authOptions, callback: { tokenRequest, error in
+                        expect(tokenRequest?.timestamp).to(beCloseTo(serverTime!, within: 6.0))
+                    })
+                }
+            }
+
+            // RSA9e
+            it("should TTL be optional and specified in milliseconds") {
+                let rest = ARTRest(options: AblyTests.commonAppSetup())
+
+                rest.auth.createTokenRequest(nil, options: nil, callback: { tokenRequest, error in
+                    //In Seconds because TTL property is a NSTimeInterval but further it does the conversion to milliseconds
+                    expect(tokenRequest?.ttl).to(equal(ARTDefault.ttl() * 60.0))
+                })
+
+                let tokenParams = ARTAuthTokenParams()
+                expect(tokenParams.ttl).to(equal(ARTDefault.ttl() * 60.0))
+
+                let expectedTtl = 10.0
+                tokenParams.ttl = expectedTtl
+
+                rest.auth.createTokenRequest(tokenParams, options: nil, callback: { tokenRequest, error in
+                    //In Seconds because TTL property is a NSTimeInterval but further it does the conversion to milliseconds
+                    expect(tokenRequest?.ttl).to(equal(expectedTtl))
+                })
+            }
+
+            // RSA9f
+            it("should provide capability has json text") {
+                let rest = ARTRest(options: AblyTests.commonAppSetup())
+
+                let tokenParams = ARTAuthTokenParams()
+                tokenParams.capability = "{ - }"
+
+                rest.auth.createTokenRequest(tokenParams, options: nil, callback: { tokenRequest, error in
+                    expect(error).toNot(beNil())
+                    expect(error?.description).to(contain("Capability"))
+                    expect(tokenRequest?.capability).to(beNil())
+                })
+
+                let expectedCapability = "{ \"cansubscribe:*\":[\"subscribe\"] }"
+                tokenParams.capability = expectedCapability
+
+                rest.auth.createTokenRequest(tokenParams, options: nil, callback: { tokenRequest, error in
+                    expect(error).to(beNil())
+                    expect(tokenRequest?.capability).to(equal(expectedCapability))
+                })
+            }
+
+            // RSA9g
+            it("should generate a valid HMAC") {
+                let rest = ARTRest(options: AblyTests.commonAppSetup())
+
+                let tokenParams = ARTAuthTokenParams(clientId: "client_string")
+
+                rest.auth.createTokenRequest(tokenParams, options: nil, callback: { tokenRequest, error in
+                    let signed = tokenParams.sign(rest.options.key!, withNonce: tokenRequest?.nonce ?? "")
```

Comment:
@mattheworiordan I don't see a way to explicitly set the nonce with the current API. Any thoughts? Or perhaps I didn't get your idea.